### DISCUSSION TITLE
Where're all the packages downloaded?

### DISCUSSION THREAD

**Discussion by jarmos-san** [tectonic-typesetting/tectonic#897] [Q&A] [answered]
`tectonic`'s feature to download all the necessary packages automatically is truly a lifesaver. But they're eating away my previous SSD space & would prefer cleaning them up after my work with LaTex is complete. So, I wondering where should I look into while searching for the packages?

I use Ubuntu but it would nice to figure out where does `tectonic` download the packages on MacOS & Windows as well for others who're interested in knowing it.

**Comment by pkgw** [comment, marked as answer]
As of recent versions you can discover this with the following command:

```
tectonic -X show user-cache-dir
```

On my computer, this prints out

```
/home/peter/.cache/Tectonic
```

This location is determined by the [app_dirs2][1] crate.

[1]: https://docs.rs/app_dirs2/latest/app_dirs2/

As of 0.9.0, the default location can be overridden with the environment variable `$TECTONIC_CACHE_DIR`. (Although it looks like the `show user-cache-dir` command doesn't pay attention to that setting, oops!)
